src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -254,10 +255,15 @@ private ICommand GetTargetCommand()
             var command = CommandFactoryUsingResolver.Create(commandSpec)
                 .WorkingDirectory(runWorkingDirectory);
 
-            var rootVariableName = Environment.Is64BitProcess ? "DOTNET_ROOT" : "DOTNET_ROOT(x86)";
-            if (Environment.GetEnvironmentVariable(rootVariableName) == null)
+            if ((TryGetTargetArchitecture(project.GetPropertyValue("RuntimeIdentifier"), out var targetArchitecture) ||
+                TryGetTargetArchitecture(project.GetPropertyValue("DefaultAppHostRuntimeIdentifier"), out targetArchitecture)) &&
+                targetArchitecture == RuntimeInformation.ProcessArchitecture)
             {
-                command.EnvironmentVariable(rootVariableName, Path.GetDirectoryName(new Muxer().MuxerPath));
+                var rootVariableName = Environment.Is64BitProcess ? "DOTNET_ROOT" : "DOTNET_ROOT(x86)";
```

Comment:
Have you considered setting `DOTNET_ROOT_arch` for net6.0+ projects? (I know it's not necessary, but would be quite nice).